### PR TITLE
Allow serialize List value

### PR DESCRIPTION
**Pull request by opodartho**

### CODE REVIEW

**Comment by doomspork** [comment]
@jeregrine / @snewcomer / @mitchellhenke / @jherdman as mentioned in #245 this PR also solves the problem but is smaller in scope of changes. I'm happy to resolve the linting issue but thoughts on this vs #245?

File: lib/jsonapi/utils/list.ex [hunk 1]
```diff
@@ -29,7 +32,7 @@ defmodule JSONAPI.Utils.List do
   end
 
   defp do_to_list_of_query_string_components({key, value}) when is_map(value) do
-    Enum.flat_map(value, fn {k, v} -> to_list_of_two_elem_tuple("#{key}[#{k}]", v) end)
+    Enum.flat_map(value, fn {k, v} -> do_to_list_of_query_string_components({"#{key}[#{k}]", v}) end)
```
Comment (doomspork) [inline lib/jsonapi/utils/list.ex:35]: This appears to be the linting issue:

```suggestion
    Enum.flat_map(value, fn {k, v} ->
      do_to_list_of_query_string_components({"#{key}[#{k}]", v})
    end)
```

**Comment by opodartho** [comment]
@doomspork as #245 also solve the same issue and change keys inside List which seems more reasonable. Feel free to close this Merge request.

**Comment by doomspork** [comment]
Thank you @opodartho :pray:

**Pull request closed without merging**